service/src/test/java/com/codeforcommunity/processor/ProtectedUserProcessorImplTest.java

```diff
@@ -0,0 +1,178 @@
+  @BeforeEach
+  void setup() {
+    this.mockDb = new JooqMock();
+    this.processor = new ProtectedUserProcessorImpl(mockDb.getContext());
+  }
+
+  // successfully deletes user when user is a member
+  @Test
+  public void testDeleteUser1() {
+    UserTeamRecord myUserTeam = mockDb.getContext().newRecord(Tables.USER_TEAM);
+    myUserTeam.setUserId(1);
+    myUserTeam.setTeamId(5);
+    myUserTeam.setTeamRole(TeamRole.MEMBER);
+    mockDb.addReturn("SELECT", myUserTeam);
+    mockDb.addEmptyReturn("DELETE");
+
+    JWTData jwtData = new JWTData(1, PrivilegeLevel.STANDARD);
+
+    this.processor.deleteUser(jwtData);
+    assertEquals(1, mockDb.timesCalled("SELECT"));
+    assertEquals(3, mockDb.timesCalled("DELETE"));
+    // user_id
+    assertEquals(1, mockDb.getSqlBindings().get("DELETE").get(0)[0]);
+    // user_id from user_team
+    assertEquals(1, mockDb.getSqlBindings().get("DELETE").get(1)[0]);
+    // user_id from users
+    assertEquals(1, mockDb.getSqlBindings().get("DELETE").get(2)[0]);
+    // user_id from user_team
+    assertEquals(1, mockDb.getSqlBindings().get("SELECT").get(0)[0]);
+  }
+
+  // successfully deletes user when user is a leader
+  @Test
+  public void testDeleteUser2() {
+    UserTeamRecord myUserTeam = mockDb.getContext().newRecord(Tables.USER_TEAM);
+    myUserTeam.setUserId(1);
+    myUserTeam.setTeamId(5);
+    myUserTeam.setTeamRole(TeamRole.LEADER);
+    mockDb.addReturn("SELECT", myUserTeam);
+    mockDb.addEmptyReturn("DELETE");
+
+    TeamRecord myTeamRecord = mockDb.getContext().newRecord(Tables.TEAM);
+    myTeamRecord.setId(5);
+    myTeamRecord.setName("Team 5");
+    myTeamRecord.setBio("bio");
+    myTeamRecord.setGoal(2);
+    Timestamp timestamp = new Timestamp(System.currentTimeMillis());
+    myTeamRecord.setGoalCompletionDate(timestamp);
+    mockDb.addReturn("SELECT", myTeamRecord);
+    mockDb.addEmptyReturn("DELETE");
+
+    JWTData jwtData = new JWTData(1, PrivilegeLevel.STANDARD);
+
+    this.processor.deleteUser(jwtData);
+    assertEquals(1, mockDb.timesCalled("SELECT"));
+    assertEquals(4, mockDb.timesCalled("DELETE"));
+    assertEquals(1, mockDb.getSqlBindings().get("DELETE").get(0)[0]);
+    assertEquals(5, mockDb.getSqlBindings().get("DELETE").get(1)[0]);
+    assertEquals(1, mockDb.getSqlBindings().get("DELETE").get(2)[0]);
+    assertEquals(5, mockDb.getSqlBindings().get("DELETE").get(3)[0]);
+    assertEquals(1, mockDb.getSqlBindings().get("SELECT").get(0)[0]);
+  }
+
+  // attempts to delete user when there are no users
+  @Test
+  public void testDeleteUser3() {
+    UsersRecord myUser = mockDb.getContext().newRecord(Tables.USERS);
+    myUser.setUsername("kiminusername");
+    myUser.setEmail("kimin@example.com");
+    myUser.setPassHash(Passwords.createHash("password"));
+    myUser.setId(1);
+    myUser.setPrivilegeLevel(PrivilegeLevel.STANDARD);
+    mockDb.addEmptyReturn("SELECT");
+    mockDb.addEmptyReturn("DELETE");
+    mockDb.addEmptyReturn("DELETE");
```

Comment:
Why are there two calls to `addEmptyReturn("DELETE")` if you're only expecting one to be made?

Reply:
I think I thought of it as attempting to deleting one for the users and one for the team

Reply:
Cool, thanks for checking on that.